### ISSUE TITLE
Should error on invalid OSM data files

### ISSUE THREAD

**Issue by danpat** [Project-OSRM/osrm-backend#1596]
We erroneously generated a `.osm` file that was missing `<node>` tags that were referred to by some `<way>`s.

Unfortunately, this took a little while to debug, because `osrm-extract` processed the file happily, but `osrm-prepare` segfaults partway through processing.  With a debug build, `osrm-prepare` throws this assertion:

https://github.com/Project-OSRM/osrm-backend/blob/develop/algorithms/graph_compressor.cpp#L85

which is unsurprising, because there are a bunch nodes missing.

We should catch this in the extractor and error out with a useful message if the original input graph is invalid.

**Comment by TheMarex** [comment]
Yeah we might want to throw an error [here](https://github.com/Project-OSRM/osrm-backend/blob/develop/extractor/extraction_containers.cpp#L189)
